### PR TITLE
[Precision Depth Alignment]  paddle.nn.functional.softplus accuracy and torch alignment

### PR DESCRIPTION
**Pull request by zhengshengning**
### PR Category

Operator Mechanism

### PR Types

Bug fixes

### Description

torch实现代码路径：src/pytorch/aten/src/ATen/native/cuda/ActivationSoftplusKernel.cu

**主要改动点（为与 PyTorch 精度对齐）：**

**前向 softplus 数值稳定性** ：
新增` log1p_local `的复数重载，将` log(1 + exp(bx))` 改为` log1p_local(exp(bx))`：
1. 实数：`log1p_local `直接调用 log1p，提升 bx 很负时的精度（避免 log(1+y) 的消失误差）。
2. 复数：自定义 `log1p_local(Complex) `为 `log(1 + exp(x))`，兼容无复杂版 log1p 的情况。

**反向梯度重写为稳定形式**
1. 实数：由 `dout / (1 + exp(-bx)) `改为设 `z=exp(bx)，dout * z / (z + 1)`，避免计算 `exp(-bx) `带来的上溢/下溢，和 PyTorch 表达一致。
2. 复数：由 `dout / conj(1 + exp(-bx)) `改为 `dout * conj(z / (z + 1))`。

**复测结果：**
paddle.nn.functional.softplus 对齐case（共29 个）
未对齐case（3个）: 未对齐的原因是函数签名不同，beta参数在paddle中只接收float类型，当beta在计算时强转成double时会出现精度问题。
```
paddle.nn.functional.softplus(Tensor([3, 3, 3],"float64"), -1e-06, 20, None, )
paddle.nn.functional.softplus(x=Tensor([3, 3, 3],"float64"), beta=1e-06, threshold=20, )
paddle.nn.functional.softplus(x=Tensor([3, 3, 3],"float64"), beta=-1e-06, threshold=20, )
```
**上面未对齐的3个double类型的case，需要改动函数签名等，所以在另一个pr中修复（https://github.com/PaddlePaddle/Paddle/pull/75426）**

pcard-67164

### CODE REVIEW

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by zrr1999 — APPROVED** [review]
